### PR TITLE
Add help feature and refactor code (resolves #3)

### LINKED ISSUE #3: Print out a descriptive help message
Plotsbot should print out a descriptive help message that helps the asker make the best possible use of the bot's functionalities. The initial help message, printed out when the user enters `@plotsbot help` should not be very long, and should rather focus on introducing the multiple different facades and how to get help for each of them, thus avoiding spamming the chatroom with the entire help document, which might contain modules not of interest to the asker.

A reference to this command (`@plotsbot help`) should be in the welcome greeting, modifying the current greeting to somewhat the likes of `Welcome to Publiclab, ${user}! Please send `@plotsbot help` for further help.`

-----

@jywarren We could also make the help feature work over DMs, thus avoiding the spamming problem in almost its complete entirety. What do you say about that?

### PR DESCRIPTION
**Pull request by ryzokuken**
This pull request introduces two refactors and one minor feature (printing help message)

### CODE REVIEW

**Comment by ryzokuken** [comment]
@publiclab/soc please look into it if possible. Any feedback is welcome.

**Comment by jywarren** [comment]
This looks great. Do you think you could link to the help message, with a one-sentence explanation, from some section of the README? This is a great first simple example of the chatbot, and deserves to be highlighted!

**Comment by ananyo2012** [comment]
@ryzokuken Is there any way I can test this out in node console?

**Comment by ryzokuken** [comment]
@ananyo2012 Maybe I could make a feature to allow you to test it out? I guess I made it in the other bot we made. 

It uses `stdin` for input/output instead of the chatroom. Would that work for you?

Meanwhile, my own instance is running right now, maybe you could test it?

**Comment by ananyo2012** [comment]
Yes it will be really great! It will allow me to more confidently merge the PRs and you can deploy it once it is merged.

**Comment by ryzokuken** [comment]
Would you like me to add the testing feature right in this branch? I could add it later on its own branch.

**Comment by ananyo2012** [comment]
Yes make an issue for it and label it testing. Can you just add this feature in readme and I can merge this!

**Comment by ryzokuken** [comment]
@jywarren I'm sorry I could not clearly understand you. You need me to add a link in the README that leads to the help message inside the codebase?

Thanks

**Comment by jywarren** [comment]
Yes, just a brief subsection that explains the help feature (as a an example of the functionality of the bot) and links to the section of code where the help feature is written. This will help newcomers understand specifically what features the bot has, and where to find them in the code. We won't have to do this for all features, but since this is one of the first ones, it's a good example to present. Make sense?

File: services/chatbot.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    if (to.charAt(0) === '#') {
+      this.client.sendMessage(to, message);
+    } else {
+      this.client.sendMessage(from, message)
+    }
+  }
+
+  addListeners () {
+    this.client.addJoinHandler((channel, nick) => {
+      this.client.sendMessage(channel, `Welcome to Publiclab, ${nick}!`)
```
Comment (jywarren) [inline services/chatbot.js]: Here we may want the bot to tell the user how to ask for help, what the correct syntax is, so they don't have to guess!

File: utils.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+function parseMessage (message) {
+  return message.split(/[\s,.;:!?]/g).filter(String);
+}
+
+function messageResponse (name, message) {
+  console.log('Bot was mentioned!');
+
+  switch (message[0]) {
```
Comment (jywarren) [inline utils.js]: Here, is there a way to accept variations, like if they put a comma in, or capitalization, or anything?

Comment (ryzokuken) [inline utils.js]: Yes, there are quite a few variations. Firstly, I removed all punctuation from the sentence and then split the string from spaces, so now I have an array of all the raw words. Then I check if either `@${botName}` or simply `${botName}` was one of the words in the list. If yes -- the bot was mentioned. The bot does not respond unless its mentioned in the sentence.

File: services/chatbot.js (hunk 1 above)
Comment (ryzokuken) [inline services/chatbot.js]: Will definitely do that.

**Comment by ananyo2012** [comment]
Well I am not sure about the module naming conventions in javascript. But I think you should put the `services/irc` module in `client/irc` module instead. That would make it easier to distinguish between the client modules and the services modules.

**Comment by ananyo2012** [comment]
This looks great! I am going ahead and merging this.

**Pull request merged**